backend/src/crd/controller/scheduledworkflow/util/cron_schedule.go

```diff
@@ -85,36 +86,39 @@ func (s *CronSchedule) getNextScheduledEpochImp(lastJobEpoch int64, catchup bool
 		return math.MaxInt64
 	}
 
-	startEpoch := lastJobEpoch
-	if s.StartTime != nil && s.StartTime.Unix() > startEpoch {
-		startEpoch = s.StartTime.Unix()
+	startTime := lastJobTime
+	if s.StartTime != nil && s.StartTime.Time.After(startTime) {
+		startTime = s.StartTime.Time
 	}
-	result := schedule.Next(time.Unix(startEpoch, 0).UTC()).Unix()
-
-	var endTime int64 = math.MaxInt64
+	result := schedule.Next(startTime)
```

Comment:
I think this is the only line (including other calls to `schedule` or `cron` package) you need to convert startTime to the location we care about, because all other operations will not differ based on timezone.

Reply:
Also, to verify your PR has the effect you want, I'd suggest creating a failing case and fix it by adding your changes. (I didn't check tests carefully, if you have already done that, that's great!)

Reply:
I have made a draft on a failing test, will continue the work and debug the test further.